### PR TITLE
feat: unflag experimental.assets

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

What the title says!

## Testing

Updated tests and they should pass!

## Docs

Sarah has a PR to docs in flight already for this!

### CODE REVIEW

**Review by matthewp — APPROVED** [review]

File: .changeset/warm-weeks-yell.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+"astro": major
+---
+
+`astro:assets` is now enabled by default and does not require `experimental.assets` to be enabled. In addition to the `astro:assets` module, this has two changes included by default:
+
+- New ESM shape, importing an image will now return an object with different properties describing the image such as its path, format and dimensions.
+- In Markdown, MDX, and Markdoc, images refered to using the `![]()` syntax will now properly be resolved no matter their position. This notably unlocks storing images next to your content.
```
Comment (sarah11918) [inline .changeset/warm-weeks-yell.md:8]:
```suggestion
`astro:assets` is now enabled by default. If you were previously using the `experimental.assets` flag, please remove it from your config. Also note that the previous `@astrojs/image` integration is incompatible, and must be removed.

This also brings two important changes to using images in Astro:

- New ESM shape: importing an image will now return an object with different properties describing the image such as its path, format and dimensions. This is a breaking change and will may require you to update your existing images.
- In Markdown, MDX, and Markdoc, the `![]()` syntax will now resolve relative images located anywhere in your project in addition to remote images and images stored in the `public/` folder. This notably unlocks storing images next to your content.

Please see our existing [Assets page in Docs](https://docs.astro.build/en/guides/assets/) for more information about using `astro:assets`.
```

**Review by sarah11918 — COMMENTED** [review]
Hey @Princesseuh ! This is exciting.... :houston_intensifies:

Since it's just the changelog for the beta, we probably don't have to get *too* deep, but I thought that at least adding in that you have to uninstall the old integration, as well as a link to docs (since this is a pretty big change!) might be helpful here. So, something kind of like below I think does a good job of balancing enough warning about the breaking changes without overloading since it's just an experimental beta.

Feel free to use from this what you like!

File: packages/astro/src/vite-plugin-inject-env-ts/index.ts [hunk 2]
```diff
@@ -50,26 +50,6 @@ export async function setUpEnvTs({
...
-			await fs.promises.writeFile(envTsPath, typesEnvContents, 'utf-8');
-			info(logging, 'assets', `Added ${bold(envTsPathRelativetoRoot)} types`);
-		} else if (
-			!settings.config.experimental.assets &&
-			typesEnvContents.includes('types="astro/client-image"')
-		) {
-			typesEnvContents = typesEnvContents.replace(
-				'types="astro/client-image"',
-				'types="astro/client"'
-			);
```
[the remark this answers is not in the document]
Comment (Princesseuh) [reply packages/astro/src/vite-plugin-inject-env-ts/index.ts:68]: Hmm, it should be mentioned in the 3.0 migration docs, but we probably should've kept this for convenience too... The right type to have in 3.0 is definitely `astro/client`, no matter what.

[the remark this answers is not in the document]
Comment (fflaten) [reply packages/astro/src/vite-plugin-inject-env-ts/index.ts:68]: Thanks. Not in [migration guide](https://deploy-preview-4166--astro-docs-2.netlify.app/en/guides/upgrade-to/v3/) yet. Not breaking builds, so might be enough to mentioned it there.

Just wasn't obvious that I had to revert before troubleshooting and reading the TODO about merging them in this PR.

**Pull request merged**